cdap-explore-client/src/main/java/co/cask/cdap/explore/client/ExploreHttpClient.java

```diff
@@ -115,35 +117,35 @@ protected boolean isAvailable() {
     }
   }
 
-  protected QueryHandle doEnableExploreStream(Id.Stream stream, String tableName,
+  protected QueryHandle doEnableExploreStream(StreamId stream, String tableName,
                                               FormatSpecification format) throws ExploreException {
     HttpResponse response = doPost(String.format(
       "namespaces/%s/data/explore/streams/%s/tables/%s/enable",
-      stream.getNamespaceId(), stream.getId(), tableName), format == null ? null : GSON.toJson(format), null);
+      stream.getNamespace(), stream.getEntityName(), tableName), format == null ? null : GSON.toJson(format), null);
```

Comment:
Does it make sense to use the entity-specific method, when you know the entity type?
For instance, use `stream.getStream()`, instead of `stream.getEntityName()`.